### ISSUE TITLE
Stream url is not stripped of piped http headers

### ISSUE THREAD

**Issue by jairoxyz** [xbmc/inputstream.ffmpegdirect#12]
When using http headers after pipe symbol, the headers are correctly and passed to ffmpeg but the url still has the headers attached to it and so it won't play. Is standard pipe symbol not the correct way to pass http headers?

```
ffmpeg[0000000000000000X]: [tcp] Successfully connected to 122.2.210.210 port 80
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] request: GET /iphone/samples/bipbop/bipbopall.m3u8|Referer=Google.Com&User-Agent=Mozilla/5.0 (iPhone; CPU iPhone OS 10_3 like Mac OS X) AppleWebKit/602.1.50 (KHTML, like Gecko) CriOS/56.0.2924.75 Mobile/14E5239e Safari/602.1 HTTP/1.1
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] User-Agent: Mozilla/5.0 (iPhone; CPU iPhone OS 10_3 like Mac OS X) AppleWebKit/602.1.50 (KHTML, like Gecko) CriOS/56.0.2924.75 Mobile/14E5239e Safari/602.1
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] Accept: */*
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] Range: bytes=0-
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] Connection: close
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] Host: devimages.apple.com
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] Icy-MetaData: 1
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] Referer: Google.Com
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http]
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] header='HTTP/1.1 503 Service Unavailable'
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] http_code=503
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] HTTP error 503 Service Unavailable
DEBUG: AddOnLog: Inputstream FFmpeg Direct: Error, could not open file http://devimages.apple.com/iphone/samples/bipbop/bipbopall.m3u8|Referer=Google.Com&User-Agent=Mozilla/5.0 (iPhone; CPU iPhone OS 10_3 like Mac OS X) AppleWebKit/602.1.50 (KHTML, like Gecko) CriOS/56.0.2924.75 Mobile/14E5239e Safari/602.1
ERROR: CVideoPlayer::OpenInputStream - error opening
```
Thanks in advance

**Comment by phunkyfish** [comment]
That may just be a bug. Let me check it.

**Comment by phunkyfish** [comment]
Yes, it’s a bug. Will fix in next release

**Comment by phunkyfish** [comment]
Can test with the artifacts from here: https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.ffmpegdirect/detail/PR-10/12/artifacts

And see if it fixes the issue?

**Comment by jairoxyz** [comment]
Wow, that was a fast response - thanks a lot. Yes, it's working perfectly now. Headers are passed correctly and stream url is correct:

```
ffmpeg[0000000000000000X]: [tcp] Starting connection attempt to 122.2.209.34 port 80
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [tcp] Successfully connected to 122.2.209.34 port 80
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] request: GET /iphone/samples/bipbop/bipbopall.m3u8 HTTP/1.1
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] User-Agent: Mozilla/5.0 (iPhone; CPU iPhone OS 10_3 like Mac OS X) AppleWebKit/602.1.50 (KHTML, like Gecko) CriOS/56.0.2924.75 Mobile/14E5239e Safari/602.1
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] Accept: */*
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] Range: bytes=0-
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] Connection: close
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] Host: devimages.apple.com
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] Icy-MetaData: 1
DEBUG: AddOnLog: Inputstream FFmpeg Direct: ffmpeg[0000000000000000X]: [http] Referer: Google.Com
```

Cheers!

**Comment by phunkyfish** [comment]
What platform do you run on? Just curious

**Comment by jairoxyz** [comment]
Just tested on Windows for now but will test on others soon, too. I mainly use Kodi on a NAS that runs on Ubuntu, RPI and FireTV.

**Comment by phunkyfish** [comment]
Cool. Let me know if there are any differences across platforms. Thanks for testing!

**Comment by phunkyfish** [comment]
Fixed in #10
